### PR TITLE
[ZEPPELIN-2001] Publish Apache Zeppelin artifacts to snapshot repository

### PR DESCRIPTION
**Pull request by anthonycorbacho**
### What is this PR for?
Simple script that deploy Apache Zeppelin artifact to snapshot repository

### What type of PR is it?
[Improvement ]

### What is the Jira issue?
 * [ZEPPELIN-2001](https://issues.apache.org/jira/browse/ZEPPELIN-2001)

### How should this be tested?
Simply run `./dev/publish_release.sh 0.7.0-SNAPSHOT branch-0.7`

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? YES

### CODE REVIEW

**Comment by anthonycorbacho** [comment]
merging if no more discussion

File: dev/publish_release.sh [hunk 1]
```diff
@@ -67,6 +70,34 @@ function curl_error() {
...
+  fi
+
+  mvn versions:set -DnewVersion=$RELEASE_VERSION
+  tmp_settings="tmp-settings.xml"
+  echo "<settings><servers><server>" > $tmp_settings
+  echo "<id>apache.snapshots.https</id><username>$ASF_USERID</username>" >> $tmp_settings
+  echo "<password>$ASF_PASSWORD</password>" >> $tmp_settings
+  echo "</server></servers></settings>" >> $tmp_settings
+
+  mvn --settings $tmp_settings -DskipTests $PUBLISH_PROFILES -Drat.skip=true deploy
```
Comment (felixcheung) [inline dev/publish_release.sh]: this seems slightly different from publish_to_maven() below - should it `-Pbeam`?  `-Dmaven.repo.local`?

**Comment by anthonycorbacho** [comment]
@felixcheung thank you for your feedback, i have updated this part, please let me know if something else need to be addressed.

**Comment by anthonycorbacho** [comment]
Merging if no more discussion 💃

**Pull request closed without merging**